### PR TITLE
feat: [CDS-50535]: Tas Route Mapping Step

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

Tas Route Mapping Step for NG. Yaml changes shown here -

https://harness.atlassian.net/wiki/spaces/CDP/pages/21211480091/PCF+NG+-+YAML
https://harness.atlassian.net/wiki/spaces/PD/pages/21210531200/PCF+NG+-+Technical+Proposal#Use-case4---Rolling-Deployment

Since this is not part of specific strategy, we do rollback in the same step.  Incase timeout occurs, the rollback might not take place. This is a limitation in NG. Pipeline team would work on it, so that we get the control incase timeout occurs

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42741)

### CODE REVIEW

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    CfRouteMappingRequestNG cfRouteMappingRequestNG =
+        CfRouteMappingRequestNG.builder()
+            .accountId(AmbianceUtils.getAccountId(ambiance))
+            .cfCommandTypeNG(CfCommandTypeNG.ROUTE_MAPPING)
+            .commandName(CfCommandUnitConstants.RouteMapping)
+            .commandUnitsProgress(CommandUnitsProgress.builder().build())
+            .timeoutIntervalInMin(CDStepHelper.getTimeoutInMin(stepParameters))
+            .useCfCLI(true)
+            .cfCliVersion(CfCliVersion.V7)
+            .mapType(tasRouteMappingStepParameters.getRouteType().equals(TasRouteType.MAP))
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:108]: can be attachRoutes

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            .routeMaps(getParameterFieldValue(tasRouteMappingStepParameters.getRoutes()))
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:109]: can be routes

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    TasInfraConfig tasInfraConfig = getTasInfraConfig(ambiance);
+
+    CfRouteMappingRequestNG cfRouteMappingRequestNG =
+        CfRouteMappingRequestNG.builder()
+            .accountId(AmbianceUtils.getAccountId(ambiance))
+            .cfCommandTypeNG(CfCommandTypeNG.ROUTE_MAPPING)
+            .commandName(CfCommandUnitConstants.RouteMapping)
+            .commandUnitsProgress(CommandUnitsProgress.builder().build())
+            .timeoutIntervalInMin(CDStepHelper.getTimeoutInMin(stepParameters))
+            .useCfCLI(true)
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:106]: check if use cli or sdk

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStepParameters.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.TypeAlias;
+
+@OwnedBy(HarnessTeam.CDP)
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeName(StepSpecTypeConstants.TAS_ROLLING_DEPLOY)
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStepParameters.java:30]: should be route_mapping

File: 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java [hunk 5]
```diff
@@ -127,7 +127,8 @@ public enum ExecutionNodeType {
   TAS_ROLLING_ROLLBACK("TAS_ROLLING_ROLLBACK", YamlTypes.TAS_ROLLING_ROLLBACK),
   ASG_BLUE_GREEN_SWAP_SERVICE("ASG_BLUE_GREEN_SWAP_SERVICE", YamlTypes.ASG_BLUE_GREEN_SWAP_SERVICE),
   ASG_BLUE_GREEN_DEPLOY("ASG_BLUE_GREEN_DEPLOY", YamlTypes.ASG_BLUE_GREEN_DEPLOY),
-  ASG_BLUE_GREEN_ROLLBACK("ASG_BLUE_GREEN_ROLLBACK", YamlTypes.ASG_BLUE_GREEN_ROLLBACK);
+  ASG_BLUE_GREEN_ROLLBACK("ASG_BLUE_GREEN_ROLLBACK", YamlTypes.ASG_BLUE_GREEN_ROLLBACK),
+  ROUTE_MAPPING("ROUTE_MAPPING", YamlTypes.TAS_ROLLING_ROLLBACK);
```
Comment (RishabhGupta34) [inline 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java:131]: yaml type should be route mapping

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java [hunk 6]
```diff
@@ -195,4 +195,5 @@ public interface YamlTypes {
   String TAS_ROLLING_ROLLBACK = StepSpecTypeConstants.TAS_ROLLING_ROLLBACK;
 
   String K8S_DRY_RUN_MANIFEST = StepSpecTypeConstants.K8S_DRY_RUN_MANIFEST;
+  String ROUTE_MAPPING = StepSpecTypeConstants.ROUTE_MAPPING;
```
Comment (RishabhGupta34) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java:198]: name change to tas_route_mapping

File: 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java [hunk 7]
```diff
@@ -2051,6 +2052,7 @@ private void bindDelegateTasks() {
     mapBinder.addBinding(TaskType.TAS_DATA_FETCH).toInstance(TasDataFetchTask.class);
     mapBinder.addBinding(TaskType.TAS_ROLLING_DEPLOY).toInstance(TasRollingDeploymentTask.class);
     mapBinder.addBinding(TaskType.TAS_ROLLING_ROLLBACK).toInstance(TasRollingRollbackTask.class);
+    mapBinder.addBinding(TaskType.ROUTE_MAPPING).toInstance(TasRouteMappingTask.class);
```
Comment (RishabhGupta34) [inline 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java:2055]: should be tas

File: 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfRouteMappingCommandTaskHandlerNG.java [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+  @Inject CfDeploymentManager cfDeploymentManager;
+  @Inject protected CfCommandTaskHelperNG cfCommandTaskHelperNG;
+  @Inject private TasRegistrySettingsAdapter tasRegistrySettingsAdapter;
+
+  @Override
+  protected CfCommandResponseNG executeTaskInternal(CfCommandRequestNG cfCommandRequestNG,
+      ILogStreamingTaskClient iLogStreamingTaskClient, CommandUnitsProgress commandUnitsProgress) throws Exception {
+    if (!(cfCommandRequestNG instanceof CfRouteMappingRequestNG)) {
+      throw new InvalidArgumentsException(
+          Pair.of("cfCommandRequestNG", "Must be instance of CfRouteMappingRequestNGx"));
```
Comment (RishabhGupta34) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfRouteMappingCommandTaskHandlerNG.java:79]: nit remove x

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/CfCommandTypeNG.java [hunk 9]
```diff
@@ -16,5 +16,6 @@ public enum CfCommandTypeNG {
   SWAP_ROLLBACK,
   DATA_FETCH,
   TAS_ROLLING_DEPLOY,
-  TAS_ROLLING_ROLLBACK
+  TAS_ROLLING_ROLLBACK,
+  ROUTE_MAPPING
```
Comment (RishabhGupta34) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/CfCommandTypeNG.java:20]: change name

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfRouteMappingRequestNG.java [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.List;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@OwnedBy(CDP)
+@EqualsAndHashCode(callSuper = true)
+public class CfRouteMappingRequestNG extends AbstractTasTaskRequest {
+  String applicationName;
```
Comment (RishabhGupta34) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfRouteMappingRequestNG.java:47]: should have allow secrets

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/response/CfRouteMappingResponseNG.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+@Data
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+public class CfRouteMappingResponseNG implements CfCommandResponseNG {
+  @NonFinal DelegateMetaInfo delegateMetaInfo;
+  @NonFinal UnitProgressData unitProgressData;
+  CommandExecutionStatus commandExecutionStatus;
+  private TasApplicationInfo currentProdInfo;
+  String errorMessage;
+  private List<CfInternalInstanceElement> newAppInstances;
```
Comment (RishabhGupta34) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/response/CfRouteMappingResponseNG.java:32]: newAppInstances, currentProdInfo can be removed

File: 970-ng-commons/src/main/java/io/harness/EntityType.java [hunk 12]
```diff
@@ -518,7 +518,11 @@ public enum EntityType {
       EntityYamlRootNames.ASG_BLUE_GREEN_DEPLOY),
   @JsonProperty(EntityTypeConstants.ASG_BLUE_GREEN_ROLLBACK)
   ASG_BLUE_GREEN_ROLLBACK_STEP(ModuleType.CD, EntityTypeConstants.ASG_BLUE_GREEN_ROLLBACK, IdentifierRef.class,
-      EntityYamlRootNames.ASG_BLUE_GREEN_ROLLBACK);
+      EntityYamlRootNames.ASG_BLUE_GREEN_ROLLBACK),
+  @JsonProperty(EntityTypeConstants.ROUTE_MAPPING)
+  ROUTE_MAPPING(ModuleType.CD, EntityTypeConstants.ROUTE_MAPPING, IdentifierRef.class,
```
Comment (RishabhGupta34) [inline 970-ng-commons/src/main/java/io/harness/EntityType.java:523]: name change

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/TasRouteMappingStepPlanCreator.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+import io.harness.pms.sdk.core.steps.io.StepParameters;
+
+import com.google.common.collect.Sets;
+import java.util.Set;
+
+@OwnedBy(HarnessTeam.CDP)
+public class TasRouteMappingStepPlanCreator extends CDPMSStepPlanCreatorV2<TasRouteMappingStepNode> {
+  @Override
+  public Set<String> getSupportedStepTypes() {
+    return Sets.newHashSet(ROUTE_MAPPING);
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/TasRouteMappingStepPlanCreator.java:26]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/TasRouteMappingStepVariableCreator.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+import io.harness.pms.sdk.core.pipeline.variables.GenericStepVariableCreator;
+
+import java.util.Collections;
+import java.util.Set;
+
+@OwnedBy(HarnessTeam.CDP)
+public class TasRouteMappingStepVariableCreator extends GenericStepVariableCreator<TasRouteMappingStepNode> {
+  @Override
+  public Set<String> getSupportedStepTypes() {
+    return Collections.singleton(StepSpecTypeConstants.ROUTE_MAPPING);
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/variables/TasRouteMappingStepVariableCreator.java:23]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingBaseStepInfo.java [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+@OwnedBy(HarnessTeam.CDP)
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@TypeAlias("tasRouteMappingBaseStepInfo")
+public class TasRouteMappingBaseStepInfo {
+  @YamlSchemaTypes({runtime})
+  @ApiModelProperty(dataType = SwaggerConstants.STRING_LIST_CLASSPATH)
+  ParameterField<List<TaskSelectorYaml>> delegateSelectors;
+  @NotNull @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) ParameterField<String> appName;
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingBaseStepInfo.java:36]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+
+  @Inject private CDFeatureFlagHelper cdFeatureFlagHelper;
+  @Inject private ExecutionSweepingOutputService executionSweepingOutputService;
+  @Inject private OutcomeService outcomeService;
+  @Inject private TasEntityHelper tasEntityHelper;
+  @Inject @Named("referenceFalseKryoSerializer") private KryoSerializer referenceFalseKryoSerializer;
+  @Inject private StepHelper stepHelper;
+  @Inject private TasStepHelper tasStepHelper;
+  @Inject private InstanceInfoService instanceInfoService;
+  public static final String COMMAND_UNIT = "Swap Rollback";
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:82]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java (hunk 1 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:108]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java (hunk 3 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:106]: Used Sdk

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStepParameters.java (hunk 4 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStepParameters.java:30]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java (hunk 6 above)
Comment (piyush-bhuwalka) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/visitor/YamlTypes.java:198]: Done

File: 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfRouteMappingCommandTaskHandlerNG.java (hunk 8 above)
Comment (piyush-bhuwalka) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfRouteMappingCommandTaskHandlerNG.java:79]: Done

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/CfCommandTypeNG.java (hunk 9 above)
Comment (piyush-bhuwalka) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/CfCommandTypeNG.java:20]: Done

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/response/CfRouteMappingResponseNG.java (hunk 11 above)
Comment (piyush-bhuwalka) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/response/CfRouteMappingResponseNG.java:32]: Done

File: 970-ng-commons/src/main/java/io/harness/EntityType.java (hunk 12 above)
Comment (piyush-bhuwalka) [reply 970-ng-commons/src/main/java/io/harness/EntityType.java:523]: Done

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfRouteMappingRequestNG.java (hunk 10 above)
Comment (piyush-bhuwalka) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfRouteMappingRequestNG.java:47]: Done

File: 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfRouteMappingCommandTaskHandlerNG.java [hunk 17]
```diff
@@ -0,0 +1,242 @@
...
+                .build();
+        logCallback.saveExecutionLog(
+            String.format("No App found with name %s", cfRouteMappingRequestNG.getApplicationName()), ERROR,
+            CommandExecutionStatus.FAILURE);
+
+        return cfRouteMappingResponseNG;
+      }
+    } catch (RuntimeException | PivotalClientApiException | IOException e) {
+      Exception sanitizedException = ExceptionMessageSanitizer.sanitizeException(e);
+      log.error(PIVOTAL_CLOUD_FOUNDRY_LOG_PREFIX + "Exception in processing PCF Rolling Deployment task [{}]",
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/CfRouteMappingCommandTaskHandlerNG.java:182]: Done

File: 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java (hunk 7 above)
Comment (piyush-bhuwalka) [reply 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java:2055]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java (hunk 5 above)
Comment (piyush-bhuwalka) [reply 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/ExecutionNodeType.java:131]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java (hunk 2 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:109]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+                                  .parameters(new Object[] {cfRouteMappingRequestNG})
+                                  .build();
+    return TaskRequestsUtils.prepareCDTaskRequest(ambiance, taskData, referenceFalseKryoSerializer,
+        Arrays.asList(CfCommandUnitConstants.RouteMapping, CfCommandUnitConstants.Wrapup),
+        TaskType.ROUTE_MAPPING.getDisplayName(),
+        TaskSelectorYaml.toTaskSelector(tasRouteMappingStepParameters.getDelegateSelectors()),
+        stepHelper.getEnvironmentType(ambiance));
+  }
+
+  private TasInfraConfig getTasInfraConfig(Ambiance ambiance) {
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingStep.java:127]: Leaving it like this for noe

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingBaseStepInfo.java [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+  @NotNull TasRouteType routeType;
+  @ApiModelProperty(dataType = SwaggerConstants.STRING_LIST_CLASSPATH) ParameterField<List<String>> routes;
```
[the remark this answers is not in the document]
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRouteMappingBaseStepInfo.java:38]: Done

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger ut6

**Comment by piyush-bhuwalka** [comment]
trigger ut7

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Review by RishabhGupta34 — APPROVED** [review]
> Reviewed 2 of 36 files at r1, 3 of 21 files at r2, 3 of 20 files at r3, 39 of 39 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42741)* status: 47 file(s) to review. 19. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by RishabhGupta34 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42741)* status: 47 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Review by RishabhGupta34 — DISMISSED** [review]
> Reviewed 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42741)* status: 48 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by piyush-bhuwalka** [comment]
trigger ut6

**Comment by piyush-bhuwalka** [comment]
trigger ut7

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Comment by piyush-bhuwalka** [comment]
trigger feature-build

**Review by yogesh-chauhan — APPROVED** [review]

**Review by RishabhGupta34 — APPROVED** [review]
> Reviewed 9 of 13 files at r6, 7 of 7 files at r8, 1 of 2 files at r10, 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42741)* status: 48 file(s) to review. Waiting on team(s): del-team.

**Comment by piyush-bhuwalka** [comment]
trigger ut4

**Review by piyush-bhuwalka — COMMENTED** [review]
> Reviewed 2 of 36 files at r1, 1 of 21 files at r2, 3 of 20 files at r3, 22 of 39 files at r4, 1 of 3 files at r5, 9 of 13 files at r6, 7 of 7 files at r8, 1 of 2 files at r10, 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42741)* status: 48 file(s) to review. Waiting on team(s): del-team.

**Pull request merged**